wolfcrypt/src/chacha20_poly1305.c

```diff
@@ -93,6 +93,9 @@ int wc_ChaCha20Poly1305_Encrypt(
     if (err != 0) return err;
 
     /* Encrypt the plaintext using ChaCha20 */
+    err = wc_Chacha_SetIV(&chaChaCtx, inIV,
```

Comment:
How was this working before without the IV being set here?

Reply:
A block of data with chacha20 is 64 bytes, CHACHA20_POLY1305_AEAD_KEYSIZE is 32 bytes in chacha20_poly1305.h, leaving 32 bytes of left over stream from creating the poly1305 key. When not handling the left over stream the next call to wc_Chacha_Process moves on to the stream created with the counter set to 1. This extra wc_Chacha_SetIV is to throw out the left over 32 bytes from counter 0 that is not used and update the counter to 1.